### PR TITLE
Add restaurant design and functionality for Assignment 1

### PR DESCRIPTION
**Pull request by cristichirtos**
This pull request contains the whole functionality of the restaurant website for the first assignment. 

The changes in this PR are:
- HTML structure of the website in index.html containing the layout of the website
- CSS website design in style.css containing all the styling
- JS implementation of the dashboard functionality with the add dish or delete dish options in script.js

You can visualize the website on GitHub pages [here](https://cristichirtos.github.io/RoR-internship-assignment-1/).

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,4 +1,86 @@
...
+    <head>
+        <title>Eureka Caffe</title>
+        <link href="./style.css" type="text/css" rel="stylesheet">
+        <script src = "./script.js">
+        </script>
+    </head>
+    <body onload = "loadElements()">
+        <header>
+            <img src="./logo.png" class = "logo"/>
+            <p class = "logotext"> Eureka<br>Caffe</p>
```
Comment (AdrianFlorian) [inline index.html]: Watch out on the spacing :-) `<p class = "logotext"> Eureka <br> Caffe </p>` even if the compiler doesn't care about it, it has to be pretty on the inside as well :-)

File: script.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    dishDiv.className = "dish";
+    let image = document.createElement("img");
+    image.className = "food-img";
+    image.src = element.imageUrl;
+    let infoDiv = document.createElement("div");
+    infoDiv.className = "info";
+    let dishName = document.createElement("p");
+    let dishCategory = document.createElement("p");
+    let dishPrice = document.createElement("p");
+    dishName.className = "dishName";
```
Comment (AdrianFlorian) [inline script.js]: I see a lot of duplicated code here, is it possible to group this code into multiple methods and simplify the logic?
See my message in #ror-internship

File: script.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        loadElementsInPage(elementList);
+    }
+}
+
+function onAddDishButtonPressed() {
+    const url = document.dashboardForm.imageurl.value;
+    const name = document.dashboardForm.name.value;
+    const category = document.dashboardForm.category.value;
+    const price = document.dashboardForm.price.value;
+    const elementList = JSON.parse(localStorage.getItem('elements') || "[]");
```
Comment (AdrianFlorian) [inline script.js]: This part (line 131 -> 137) could go in a new method, what do you think?

**Review by AdrianFlorian — CHANGES_REQUESTED** [review]
Nice job Cristi 💪👏 , just a few things in general:
- avoid duplicated code by using multiple methods
- methods that do too many things should be broken down into smaller methods with a descriptive name
- How to write better commit messages: https://chris.beams.io/posts/git-commit/

**Comment by AdrianFlorian** [comment]
@cristichirtos also, let's work on a better Pull Request name - the PR name should contain a quick overview of the changes that your code contain :-)

File: script.js (hunk 3 above)
Comment (cristichirtos) [reply script.js]: I think so too, I'm working on it. Thanks!

File: index.html (hunk 1 above)
Comment (cristichirtos) [reply index.html]: Missed that one, thank you!

File: script.js (hunk 2 above)
Comment (cristichirtos) [reply script.js]: Working on a better solution right now.

File: index.html [hunk 4]
```diff
@@ -1,4 +1,86 @@
...
+            <nav>
+                <ul>
+                    <li>HOME</li>
+                    <li>ABOUT</li> 
+                    <li>SERVICE</li>
+                    <li>BLOG</li>
+                    <li>CONTACTS</li>
```
Comment (chioreandan) [inline index.html]: Indentation for your project should be 2 spaces instead of 4

File: index.html (hunk 1 above)
Comment (chioreandan) [reply index.html]: @cristichirtos In HTML you don't need to add a space `class` and `=`. 
`<p class="logotext"> Eureka <br> Caffe </p>`

Comment (chioreandan) [reply index.html]: That way is easier to read

**Comment by cristichirtos** [comment]
I have implemented the request changes and updated the PR name and description. I'm here if you see more things that can be improved!

File: style.css [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+#dashboard {
+    margin-left: 150px;
+    margin-top: 50px;
+    margin-bottom: 20px;
+    font-size: 18px;
+    text-transform: uppercase;
+}
+
+form {
```
Comment (chioreandan) [inline style.css]: Don't override styling of native components. The best approach is to create a wrapper class and add the styling there. If you need 2 form components with different styling in the same page you will have the same styling and you don't want that.

File: script.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    elementList.push(element);
+    uploadToLocalStorage(elementList);
+    loadElementInHTML(element);
+}
+
+function removeAllChildNodes(parent) {
+    while (parent.firstChild) {
+        parent.removeChild(parent.firstChild);
+    }
+}
```
Comment (chioreandan) [inline script.js]: add an empty line at the end of each file

**Review by chioreandan — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (cristichirtos) [reply index.html]: I enjoy it with spaces more, but if it improves readability in HTML, I will modify them.

**Comment by cristichirtos** [comment]
@AdrianFlorian @chioreandan I implemented the changes you have requested. Let me know what you think!

File: script.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
+function loadElements() {
+    const elementList = JSON.parse(localStorage.getItem('elements') || "[]");
+    if (elementList != null && elementList.length > 0) {
+        console.log("loading from local storage");
```
Comment (AdrianFlorian) [inline script.js]: Logs should only be used in development, please remove this one when pushing changes :-)

File: script.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        loadElementInHTML(element);
+    });
+}
+
+function loadElementInHTML(element) {
+    const containerParent = document.getElementById("container");
+    containerParent.appendChild(createDishElement(element));
+}
+
+function createDishElement(element) {
```
Comment (AdrianFlorian) [inline script.js:24]: nice 👍

**Review by AdrianFlorian — CHANGES_REQUESTED** [review]
nice! Just one thing about the log and you're good to go from my end :-)

File: script.js (hunk 7 above)
Comment (cristichirtos) [reply script.js]: I forgot the logs, thanks for the heads up!

File: script.js (hunk 8 above)
Comment (cristichirtos) [reply script.js:24]: Thank you!

**Review by AdrianFlorian — APPROVED** [review]
Nice 👏

File: style.css [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+}
+
+main .optionsBar {
+    width: 100%;
+    height: 50px;
+    background-color: #F0F0F2;
+    display: grid;
+    grid-template-columns: repeat(30, 1fr);
+    align-items: center;
+}
```
Comment (chioreandan) [inline style.css]: As a rule, it's not ok to override the native HTML components like header, nav, main, etc. 

> Dan Chiorean  12:19 PM
Inca o regula generala. Incercati sa nu puneti styling pe componente HTML native cum ar fi form, nav, img, etc  pentru ca e posibil sa aveti nevoie sa folositi aceasi componenta in locuri diferite dar cu styling diferit si atunci o sa va faca ovverride peste tot. Incercati sa folositi clase pentru a face styling-ul pe componenta respectiva

**Review by chioreandan — CHANGES_REQUESTED** [review]
Looks good @cristichirtos,
Just a small request change and you are good to go.

File: style.css (hunk 9 above)
Comment (cristichirtos) [reply style.css]: I added a class attribute for the header element, but not for the main element, as I only override an element inside of it, not main itself. Is that alright?

File: index.html [hunk 10]
```diff
@@ -1,4 +1,85 @@
 <!DOCTYPE html>
 <html>
-  <body>This is the creation of the index.html file on GitHub!</body>
+  <head>
+    <title>Eureka Caffe</title>
+    <link href="./style.css" type="text/css" rel="stylesheet">
+    <script src="./script.js">
+    </script>
+  </head>
```
Comment (chioreandan) [inline index.html:8]: Try to add empty lines after this kind of sections

**Review by chioreandan — COMMENTED** [review]
1 small comment 💯

File: index.html (hunk 10 above)
Comment (cristichirtos) [reply index.html:8]: I added spaces after all major elements in the HTML file. What do you think?

File: script.js [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+    dishContainerDiv.appendChild(image);
+    dishContainerDiv.appendChild(createInfoElement(element));
+    return dishContainerDiv;
+}
+
+function createInfoElement(element) {
+    const infoContainerDiv = createHTMLElement("div", "info");
+    const name = createHTMLElement("p", "dishName");
+    const category = createHTMLElement("p", "category");
+    const price = createHTMLElement("p", "price");
```
Comment (chioreandan) [inline script.js:39]: add space after constants declaration

File: script.js [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+    name.innerHTML = element.dishName;
+    category.innerHTML = element.category;
+    price.innerHTML = element.price;
+    infoContainerDiv.appendChild(name);
+    infoContainerDiv.appendChild(category);
+    infoContainerDiv.appendChild(price);
```
Comment (chioreandan) [inline script.js:46]: Add empty line after method actions

File: script.js [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+    return infoContainerDiv;
+}
+
+function createHTMLElement(elementType, elementClass) {
+    const newHTMLElement = document.createElement(elementType);
+    newHTMLElement.className = elementClass;
+    return newHTMLElement;
```
Comment (chioreandan) [inline script.js]: same as above

File: script.js [hunk 14]
```diff
@@ -0,0 +1,155 @@
+function loadElements() {
+    const elementList = JSON.parse(localStorage.getItem('elements') || "[]");
```
Comment (chioreandan) [inline script.js:2]: add empty line after const declaration

File: script.js [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+
+function loadElementsInPage(elements) {
+    removeAllChildNodes(document.getElementById("container"));
+    elements.forEach(function (element) {
+        loadElementInHTML(element);
+    });
+}
+
+function loadElementInHTML(element) {
+    const containerParent = document.getElementById("container");
```
Comment (chioreandan) [inline script.js:19]: add space after constants declaration

File: script.js [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+    containerParent.appendChild(createDishElement(element));
+}
+
+function createDishElement(element) {
+    const dishContainerDiv = createHTMLElement("div", "dish");
+    const image = createHTMLElement("img", "food-img");
```
Comment (chioreandan) [inline script.js:26]: add space after constants declaration

File: script.js [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+    image.src = element.imageUrl;
+    dishContainerDiv.appendChild(image);
+    dishContainerDiv.appendChild(createInfoElement(element));
```
Comment (chioreandan) [inline script.js:30]: add space after method actions

**Review by chioreandan — CHANGES_REQUESTED** [review]
Please apply the changes everywhere it is applicable

**Comment by cristichirtos** [comment]
@chioreandan I added all the spaces after constants and after the actions of functions in script.js. Let me know what you think!

**Review by chioreandan — APPROVED** [review]
Nice job @cristichirtos

**Comment by cristichirtos** [comment]
Thank you!

**Pull request merged**